### ISSUE TITLE
explicitly check for empty values when allowEmptyValues is false

### ISSUE THREAD

**Issue by karenetheridge** [karenetheridge/OpenAPI-Modern#62]
Mojo param parsing behaves as if allowEmptyValue is true, which is not the openapi default:

```
for my $pair (split /&/, $str) {
  next unless $pair =~ /^([^=]+)(?:=(.*))?$/;
  my ($name, $value) = ($1, $2 // '');
...
```

If we want to verify that empty values aren't passed, then we need to parse the parameters ourselves and error when $2 is undef.

**Comment by karenetheridge** [comment]
Just as in #63, this is difficult to do without having access to the original request (Mojo::Parameters::pairs turns a missing query value into '').

**Comment by karenetheridge** [comment]
`allowEmptyValue` recently saw some clarification in the spec: https://github.com/OAI/OpenAPI-Specification/pull/3835

..which is reflected here via commit 061d1641a3, which was released in v0.065.

**Comment by karenetheridge** [comment]
`allowEmptyValue` recently saw some clarification in the spec: https://github.com/OAI/OpenAPI-Specification/pull/3835

..which is reflected here via commit 061d1641a3, which was released in v0.065.
